File: Agregacion/ejercicio 7/ej7.py
class Estudiante:
    def __init__(self, nombre, carrera, semestre):  
        self._nombre = nombre
        self._carrera = carrera
        self._semestre = semestre

    @property
    def nombre(self):
        return self._nombre

    @nombre.setter
    def nombre(self, valor):
        self._nombre = valor

    @property
    def carrera(self):
        return self._carrera

    @carrera.setter
    def carrera(self, valor):
        self._carrera = valor

    @property
    def semestre(self):
        return self._semestre

    @semestre.setter
    def semestre(self, valor):
        self._semestre = valor

    def mostrar_info(self):
        print(f"{self.nombre} | Carrera: {self.carrera} | Semestre: {self.semestre}")

class Universidad:
    def __init__(self, nombre):  
        self._nombre = nombre
        self._estudiantes = []

    @property
    def nombre(self):
        return self._nombre

    def agregar_estudiante(self, estudiante):
        if isinstance(estudiante, Estudiante):
            self._estudiantes.append(estudiante)

    def mostrar_universidad(self):
        print(f"\nUniversidad: {self.nombre}")
        for estudiante in self._estudiantes:
            estudiante.mostrar_info()

if __name__ == "__main__":  
    u = Universidad("Universidad Mayor de San Andres")
    e1 = Estudiante("Ana", "Ingeniería", 3)
    e2 = Estudiante("Luis", "Medicina", 5)
    e3 = Estudiante("Marta", "Derecho", 2)

    u.agregar_estudiante(e1)
    u.agregar_estudiante(e2)
    u.agregar_estudiante(e3)

    u.mostrar_universidad()

File: Agregacion/ejercicio 7/ej7.java
import java.util.ArrayList;

class Estudiante {
    private String nombre;
    private String carrera;
    private int semestre;

    public Estudiante(String nombre, String carrera, int semestre) {
        this.nombre = nombre;
        this.carrera = carrera;
        this.semestre = semestre;
    }

    public String getNombre() { return nombre; }
    public void setNombre(String nombre) { this.nombre = nombre; }

    public String getCarrera() { return carrera; }
    public void setCarrera(String carrera) { this.carrera = carrera; }

    public int getSemestre() { return semestre; }
    public void setSemestre(int semestre) { this.semestre = semestre; }

    public void mostrarInfo() {
        System.out.println(nombre + " | Carrera: " + carrera + " | Semestre: " + semestre);
    }
}

class Universidad {
    private String nombre;
    private ArrayList<Estudiante> estudiantes;

    public Universidad(String nombre) {
        this.nombre = nombre;
        this.estudiantes = new ArrayList<>();
    }

    public String getNombre() { return nombre; }

    public void agregarEstudiante(Estudiante estudiante) {
        estudiantes.add(estudiante);
    }

    public void mostrarUniversidad() {
        System.out.println("\nUniversidad: " + nombre);
        for (Estudiante e : estudiantes) {
            e.mostrarInfo();
        }
    }
}

public class Main {
    public static void main(String[] args) {
        Universidad u = new Universidad("Universidad Mayor de San Andres");
        Estudiante e1 = new Estudiante("Ana", "Ingeniería", 3);
        Estudiante e2 = new Estudiante("Luis", "Medicina", 5);
        Estudiante e3 = new Estudiante("Marta", "Derecho", 2);

        u.agregarEstudiante(e1);
        u.agregarEstudiante(e2);
        u.agregarEstudiante(e3);

        u.mostrarUniversidad();
    }
}
